statistics/column.go

```diff
@@ -165,29 +158,29 @@ func (c *Column) IsInvalid(
 	}
 	if sctx != nil {
 		stmtctx := sctx.GetSessionVars().StmtCtx
-		if (!c.IsStatsInitialized() || c.IsLoadNeeded()) && stmtctx != nil {
-			if stmtctx.StatsLoad.Timeout > 0 {
-				logutil.BgLogger().Warn("Hist for column should already be loaded as sync but not found.",
-					zap.String(strconv.FormatInt(c.Info.ID, 10), c.Info.Name.O))
-			}
-			// In some tests, the c.Info is not set, so we add this check here.
-			// When we are using stats from PseudoTable(), the table ID will possibly be -1.
-			// In this case, we don't trigger stats loading.
-			if c.Info != nil && c.PhysicalID > 0 {
-				HistogramNeededItems.insert(model.TableItemID{TableID: c.PhysicalID, ID: c.Info.ID, IsIndex: false})
-			}
+		if (colStats == nil || !colStats.IsStatsInitialized() || colStats.IsLoadNeeded()) &&
+			stmtctx != nil &&
+			histColl.PhysicalID > 0 &&
+			!histColl.CanNotTriggerLoad {
+			HistogramNeededItems.insert(model.TableItemID{TableID: histColl.PhysicalID, ID: cid, IsIndex: false})
 		}
```

Comment:
Why are the logs and comments removed?

Reply:
Should be mistake when resolving merging conflicts

Reply:
That check became the condition `histColl.CanNotTriggerLoad`.